library/panic_unwind/Cargo.toml

```diff
@@ -15,11 +15,11 @@ doc = false
 alloc = { path = "../alloc" }
 core = { path = "../core" }
 unwind = { path = "../unwind" }
-compiler_builtins = "0.1.0"
+compiler_builtins = "0.1.133"
 cfg-if = { version = "1.0", features = ['rustc-dep-of-std'] }
 
 [target.'cfg(not(all(windows, target_env = "msvc")))'.dependencies]
-libc = { version = "0.2", default-features = false }
+libc = { version = "0.2.159", default-features = false }
```

Comment:
We want to pin these crates in only one place to reduce the number of places that need to be updated when bumping them. So `libc` should be left as-is, `compiler_builtins` can be left as-is or turned into only `= "0.1"`. (This applies a few places).

We could probably also use workspace dependencies to make things more clear, but this is probably something that should be brought up on Zulip first since it could affect the ability of the crates to build in different workspaces.

Reply:
The library is its own workspace now, isn't it?

Reply:
Yeah, I was mostly thinking about crates that wind up being part of different workspaces at different times through submodules or symlinks. Miri has some unusual things like that but I do doubt there is anything relevant in library/.

Reply:
I was looking into this a bit more and found this https://github.com/rust-lang/rust/commit/1f3be75f56bfa7520b86eada306ad66455b4fd6e#diff-02a067037f96954a7782700c9e368aca6229b3d8909031ec77fd67f035914853

```
While cargo workspaces normally enable dependencies of multiple targets
to be reused, for the standard library we do not want this reusing to
prevent conflicts between dependencies of the sysroot and of tools that
are built using this sysroot. For this reason we already use an unstable
cargo feature to ensure that any dependencies which would otherwise be
shared get a different -Cmetadata argument as well as using separate
build dirs.
```

It would be nice if we could figure this out somehow

Reply:
I just read through the post and I feel it would actually be nice. If it didn't fail on the CI / PR - x86_64-gnu-llvm-17, can it be approved.

Reply:
My comment here still applies https://github.com/rust-lang/rust/pull/131574#discussion_r1797508016, please change that back. Other than that I think the change looks fine (but it needs a rebase)